src/pkcs15init/profile.c

```diff
@@ -1985,24 +1985,40 @@ static struct block	root_ops = {
    "root",		process_block,	NULL,		root_blocks
 };
 
+static void
+get_inner_word(char *str, char word[64]) {
+	char *inner = NULL;
+	size_t len = 0;
+
+	inner = str;
+	while (isalnum((unsigned char)*inner) || *inner == '-' || *inner == '_') {
+		inner++;
+		len++;
+	}
+	len = len >= 64 ? 63 : len;
+	memcpy(word, str, len);
+	word[len] = '\0';
+}
+
 static int
-check_macro_reference_loop(char *start, char *str, struct state *cur, int depth) {
+check_macro_reference_loop(const char *start_name, char *value, sc_profile_t *profile, int depth) {
```

Comment:
I would propose a short doc string here with at least information what return values we can expect in which cases.
Ideally introducing also some unit tests to check the corner cases if possible as it is getting more complicated.